overlord/snapshotstate/snapshotstate.go

```diff
@@ -197,14 +229,38 @@ func Restore(st *state.State, setID uint64, snapNames []string, users []string)
 
 	ts = state.NewTaskSet()
 
-	for i, name := range snapsFound {
-		desc := fmt.Sprintf("Restore data of snap %q from snapshot set #%d", name, setID)
+	for _, summary := range summaries {
+		if summary.snapID == "" {
+			// snapshotted snap was unasserted, never mind
+			continue
+		}
+		snapst, ok := all[summary.snap]
```

Comment:
I'm wondering if it's better to use individual `snapshot.Get()` or pay the price of unmarshalling all snaps upfront.

Reply:
assuming you meant `snapstate.Get()`, perhaps yes. Doing the `All` makes the worst case better, at the expense of the expected common case. Not sure.

I'm hoping to revisit all the `snapstate.All` calls when we have Go 1.7+, because [reasons](https://gist.github.com/chipaca/10ecdf364db962c9b5ee590f0f610b2a).